alchemy-web/src/content/docs/blog/2025-08-05-alchemy-vite-plugin.md

```diff
@@ -0,0 +1,207 @@
+---
+title: Alchemy Vite Plugin
+excerpt: Bye-bye wrangler.json and .dev.vars, hello alchemy()
+date: 2025-08-05
+author: Sam Goodwin
+---
+
+Alchemy now ships with plugins for Vite, Astro, SvelteKit, Nuxt, React Router, and TanStack Start that streamline local development by eliminating the need for a `.dev.vars` file, configuration of wrangler state paths, and other boilerplate.
+
+:::note
+__TLDR__: update your `vite.config.ts` file to use the `alchemy` plugin instead of the `cloudflare` plugin for a smoother experience:
+
+```diff lang='ts'
+import { defineConfig } from "vite";
+-import { cloudflare } from "@cloudflare/vite-plugin";
++import alchemy from "alchemy/cloudflare/vite";
+
+export default defineConfig({
+-  plugins: [cloudflare()],
++  plugins: [alchemy()],
+});
+```
+:::
+
+## What problem does it solve?
+
+When deploying a Web app in Alchemy, you set secrets using the `bindings` property:
+
+```ts
+await Vite("website", {
+  bindings: {
+    SOME_SECRET: alchemy.secret.env.SOME_SECRET,
+  }
+})
+```
+
+👍 The `alchemy deploy` command will build and deploy your website to Cloudflare with the `SOME_SECRET` as a secure Binding, as expected.
+
+👎 The `alchemy dev` command, however, would be missing the secret because Cloudflare's Vite plugin only includes environment variables from `.dev.vars`.
+
+## The old workaround
+
+The workaround was to replicate these values to the `.dev.vars` file like so:
+
+```sh
+VITE_SOME_VAR=some-value
+SOME_SECRET=some-secret
+```
+
+But, this is redundant and inflexible. As Alchemy users, we want to maintain our configuration and environments in code.
+
+## The new solution
+
+We updated the `Website` resources (and its variants like `Vite`, `Astro`, etc.) to generate a temporary `wrangler.json` file in `.alchemy/local/wrangler.json` that contains your secrets in plain text:
+
+```diff lang='json'
+{
+  "name": "website",
+  "main": "./src/worker.ts",
+  "compatibility_date": "2025-08-02",
+  "assets": { "directory": "./dist/client", "binding": "ASSETS" },
++  "vars": { "SOME_SECRET": "super secret value!" },
+}
+```
+
+`vite dev` will include these secrets when running your Worker locally, so you no longer need a `.dev.vars` file 🎉.
+
+:::caution
+Ensure that `.alchemy` is added to your `.gitignore` to avoid accidentally committing secrets to your repo.
+:::
```

Comment:
Is this worth automating?

Reply:
I think we do in the templates. Might be worth adding a validation but would require we process gitignore and walk the tree to the root of the project. Can be quite complex?